src/Extension/AccountReset/SecurityExtension.php

```diff
@@ -159,21 +143,57 @@ public function doResetAccount(array $data, Form $form): HTTPResponse
         /** @var Member&MemberExtension $member */
         $member = Member::get()->byID((int) $memberID);
 
+        // Fail if Password[_Password] is missing
+        $password = $data['Password']['_Password'] ?? null;
+        if (!$password) {
+            // Using sessionMessage() rather than setFieldMessage() becaues with that method
+            // the message will not show on the field when specifying the field name as either
+            // "Password[_Password]" or even simply "Password"
+            // This is probably due to the use of ConfirmedPasswordField which isn't a standard
+            // CompositeField
+            $form->sessionMessage(
+                _t(
+                    'SilverStripe\\Forms\\Form.FIELDISREQUIRED',
+                    '{name} is required',
+                    [
+                        'name' => _t('SilverStripe\\Security\\Member.PASSWORD', 'Password'),
+                    ]
+                ),
+                ValidationResult::TYPE_ERROR
+            );
+            return $this->owner->redirectBack();
+        }
+
+        // Fail if Password[_ConfirmPassword] is missing
+        $confirmPassword = $data['Password']['_ConfirmPassword'] ?? null;
+        if (!$confirmPassword) {
+            $form->sessionMessage(
+                _t(
+                    'SilverStripe\\Forms\\Form.FIELDISREQUIRED',
+                    '{name} is required',
+                    [
+                        'name' => _t('SilverStripe\\Security\\Member.CONFIRMPASSWORD', 'Confirm Password'),
+                    ]
+                ),
+                ValidationResult::TYPE_ERROR
+            );
+            return $this->owner->redirectBack();
+        }
```

Comment:
```suggestion
```
There's no need for any of this, because `ConfirmedPasswordField` has its own validation against empty values.

Reply:
Removed